cmd/fetch.go

```diff
@@ -476,3 +504,54 @@ func fetchAvailCapacity(cmd *cobra.Command, args []string) {
 		fmt.Println(capacity)
 	}
 }
+
+//fetchTasksWithStatus returns lists of tasks for a given set of status
+func fetchTasksWithStatus(cmd *cobra.Command, args []string) {
+	status := *taskStatus
+
+	log.Infof("Fetching tasks for environment/role/job:[%s/%s/%s] \n", *env, *role, *name)
+	log.Infof("Fetching tasks for a given status: %v \n", status)
+
+	// This Query takes nil for values it shouldn't need to match against.
+	// This allows us to potentially avoid expensive calls for specific environments, roles, or job names.
+	if *env == "" {
+		env = nil
+	}
+	if *role == "" {
+		role = nil
+	}
+	if *name == "" {
+		name = nil
+	}
+	queryStatuses, err := scheduleStatusFromString(status)
+	if err != nil {
+		log.Fatalf("error: %+v", err)
+	}
+
+	taskQuery := &aurora.TaskQuery{Environment: env, Role: role, JobName: name, Statuses: queryStatuses}
+
+	tasks, err := client.GetTasksWithoutConfigs(taskQuery)
+	if err != nil {
+		log.Fatalf("error: %+v", err)
+	}
+
+	if toJson {
```

Comment:
I think we just need print out a list of task id like
role-env-name-[instance-id]

Reply:
Saw in fetchTasksConfig just do internal.ToJSON(tasks), line 260

Reply:
can you share with me the output example?

Reply:
/private/var/folders/p4/xg62qpjd17jc8wd_vsv8zww80000gp/T/GoLand/___go_build_main_go fetch tasksWithStatus -z geninmm01005.gpf-dev.us-central1.gcp.dev.paypalinc.com -x pending --toJSON
INFO[0000] Scheduler URL from ZK:  http://geninmm01005.gpf-dev.us-central1.gcp.dev.paypalinc.com:8081 
INFO[0000] gorealis clientConfig url: http://geninmm01005.gpf-dev.us-central1.gcp.dev.paypalinc.com:8081/api 
INFO[0000] Fetching tasks for environment/role/job:[//]  
INFO[0000] Fetching tasks for a given status: pending   
{"PENDING":[{"assignedTask":{"taskId":"admin-genesisinfra-apirouter-a-1-3b8122aa-35b2-4ebb-8d58-e0fa937711b6","slaveId":"","slaveHost":"","task":{"isService":true,"priority":0,"maxTaskFailures":0,"owner":{"user":"admin"},"production":false,"constraints":[{"name":"host","constraint":{"limit":{"limit":1}}},{"name":"dedicated","constraint":{"value":{"negated":false,"values":["admin/adminjob"]}}}],"executorConfig":{"name":"dce-go-paypal-host","data":""},"metadata":[{"key":"com.genesis.version","value":"genesisapiserv-v3.11.4-v2.3.5"},{"key":"genesisapi.env.SVCPORT","value":"8100"},{"key":"com.genesis.slbpool","value":"true"},{"key":"com.admin.poolname","value":"apirouter"},{"key":"com.admin.apirouter.svcport","value":"8100"},{"key":"com.genesis.pool","value":"apirouter"},{"key":"com.genesis.hostName","value":""},{"key":"com.genesis.svcport","value":"8100"},{"key":"version","value":"genesisapiserv-v3.11.4-v2.3.5"},{"key":"com.admin.apirouter.poolservercount","value":"3"},{"key":"fetch_token","value":"true"},{"key":"deployStrategy","value":"DEPLOY_STRATEGY_ROLLING_UPDATE"},{"key":"com.admin.apirouter.version","value":"genesisapiserv-v3.11.4-v2.3.5"},{"key":"genesisapi.env.FSM_COMPOSE_FILENAMES","value":"docker-compose.yaml,config/docker-compose-base.yml,config/docker-compose-extends-base.yml"},{"key":"resourceLimit.gpu.required","value":"false"},{"key":"fileName","value":"docker-compose.yml,config/docker-compose-base.yml,config/docker-compose-extends-base.yml"},{"key":"keymaker_url","value":"https://keymakerapi-vip.qa.paypal.com"},{"key":"genesisapi.env.pool","value":"apirouter"},{"key":"genesisapi.env.DZENV","value":"genesisinfra"},{"key":"com.genesis.containerIP","value":""},{"key":"requuid","value":"93130a55-cbb4-11eb-9f10-42010ab4496b"},{"key":"namespace","value":"genesisinfra"},{"key":"keymaker_skipverify","value":"1"},{"key":"nsvip","value":"10.180.82.129"},{"key":"dz","value":"genesisinfra"},{"key":"genesisapi.env.DZ","value":"gcp"},{"key":"genesisapi.env.tenant","value":"admin"},{"key":"pool","value":"apirouter"},{"key":"com.admin.apirouter.namespace","value":"genesisinfra"},{"key":"clustername","value":"avi-controller"},{"key":"com.genesis.namespace","value":"genesisinfra"},{"key":"genesisapi.env.FSM_RESOURCE_URLS","value":"http://genesisinfra.qa.paypal.com:8100/config/fsm/docker-compose.yaml http://manifestserv.paypalcorp.com/containerspecs/specid/genesisconfig-v2.3.5-fsm/config.zip"},{"key":"com.genesis.cluster","value":"genesisinfra"},{"key":"com.admin.apirouter.pooluuid","value":"pool-3d9ffffe-a7aa-40f2-8e9c-7126c75690b2"},{"key":"com.genesis.nsvip","value":"10.180.82.129"},{"key":"secretProvider","value":"keymaker"},{"key":"tenant","value":"admin"},{"key":"genesisapi.env.namespace","value":"genesisinfra"},{"key":"com.genesis.dz","value":"genesisinfra"},{"key":"com.genesis.tenant","value":"admin"},{"key":"geo","value":""}],"job":{"role":"admin","environment":"genesisinfra","name":"apirouter-a"},"container":{"mesos":{}},"tier":"preemptible","resources":[{"ramMb":1024},{"numCpus":1},{"diskMb":100},{"namedPort":"org.apache.aurora.port.0"},{"namedPort":"org.apache.aurora.port.1"},{"namedPort":"org.apache.aurora.port.2"},{"namedPort":"org.apache.aurora.port.3"},{"namedPort":"org.apache.aurora.port.4"}],"mesosFetcherUris":[{"value":"http://manifestserv.paypalcorp.com/containerspecs/genesisapiserv-v3.11.4/yaml","extract":true,"cache":false},{"value":"http://manifestserv.paypalcorp.com/containerspecs/specid/genesisconfig-v2.3.5-apirouter/config.zip","extract":true,"cache":false}]},"assignedPorts":{},"instanceId":1},"status":"PENDING","failureCount":1078,"taskEvents":[{"timestamp":1655896704650,"status":"PENDING","message":"Rescheduled","scheduler":"geninmm01006.gpf-dev.us-central1.gcp.dev.paypalinc.com"}],"ancestorId":"admin-genesisinfra-apirouter-a-1-3f20ab8d-196d-4fdf-ba90-853027680b89","timesPartitioned":0},{"assignedTask":{"taskId":"admin-genesisinfra-apirouter-a-0-8355edb8-c991-476e-9373-e325a25e6e74","slaveId":"","slaveHost":"","task":{"isService":true,"priority":0,"maxTaskFailures":0,"owner":{"user":"admin"},"production":false,"constraints":[{"name":"host","constraint":{"limit":{"limit":1}}},{"name":"dedicated","constraint":{"value":{"negated":false,"values":["admin/adminjob"]}}}],"executorConfig":{"name":"dce-go-paypal-host","data":""},"metadata":[{"key":"com.genesis.version","value":"genesisapiserv-v3.11.4-v2.3.5"},{"key":"genesisapi.env.SVCPORT","value":"8100"},{"key":"com.genesis.slbpool","value":"true"},{"key":"com.admin.poolname","value":"apirouter"},{"key":"com.admin.apirouter.svcport","value":"8100"},{"key":"com.genesis.pool","value":"apirouter"},{"key":"com.genesis.hostName","value":""},{"key":"com.genesis.svcport","value":"8100"},{"key":"version","value":"genesisapiserv-v3.11.4-v2.3.5"},{"key":"com.admin.apirouter.poolservercount","value":"3"},{"key":"fetch_token","value":"true"},{"key":"deployStrategy","value":"DEPLOY_STRATEGY_ROLLING_UPDATE"},{"key":"com.admin.apirouter.version","value":"genesisapiserv-v3.11.4-v2.3.5"},{"key":"genesisapi.env.FSM_COMPOSE_FILENAMES","value":"docker-compose.yaml,config/docker-compose-base.yml,config/docker-compose-extends-base.yml"},{"key":"resourceLimit.gpu.required","value":"false"},{"key":"fileName","value":"docker-compose.yml,config/docker-compose-base.yml,config/docker-compose-extends-base.yml"},{"key":"keymaker_url","value":"https://keymakerapi-vip.qa.paypal.com"},{"key":"genesisapi.env.pool","value":"apirouter"},{"key":"genesisapi.env.DZENV","value":"genesisinfra"},{"key":"com.genesis.containerIP","value":""},{"key":"requuid","value":"93130a55-cbb4-11eb-9f10-42010ab4496b"},{"key":"namespace","value":"genesisinfra"},{"key":"keymaker_skipverify","value":"1"},{"key":"nsvip","value":"10.180.82.129"},{"key":"dz","value":"genesisinfra"},{"key":"genesisapi.env.DZ","value":"gcp"},{"key":"genesisapi.env.tenant","value":"admin"},{"key":"pool","value":"apirouter"},{"key":"com.admin.apirouter.namespace","value":"genesisinfra"},{"key":"clustername","value":"avi-controller"},{"key":"com.genesis.namespace","value":"genesisinfra"},{"key":"genesisapi.env.FSM_RESOURCE_URLS","value":"http://genesisinfra.qa.paypal.com:8100/config/fsm/docker-compose.yaml http://manifestserv.paypalcorp.com/containerspecs/specid/genesisconfig-v2.3.5-fsm/config.zip"},{"key":"com.genesis.cluster","value":"genesisinfra"},{"key":"com.admin.apirouter.pooluuid","value":"pool-3d9ffffe-a7aa-40f2-8e9c-7126c75690b2"},{"key":"com.genesis.nsvip","value":"10.180.82.129"},{"key":"secretProvider","value":"keymaker"},{"key":"tenant","value":"admin"},{"key":"genesisapi.env.namespace","value":"genesisinfra"},{"key":"com.genesis.dz","value":"genesisinfra"},{"key":"com.genesis.tenant","value":"admin"},{"key":"geo","value":""}],"job":{"role":"admin","environment":"genesisinfra","name":"apirouter-a"},"container":{"mesos":{}},"tier":"preemptible","resources":[{"ramMb":1024},{"numCpus":1},{"diskMb":100},{"namedPort":"org.apache.aurora.port.0"},{"namedPort":"org.apache.aurora.port.1"},{"namedPort":"org.apache.aurora.port.2"},{"namedPort":"org.apache.aurora.port.3"},{"namedPort":"org.apache.aurora.port.4"}],"mesosFetcherUris":[{"value":"http://manifestserv.paypalcorp.com/containerspecs/genesisapiserv-v3.11.4/yaml","extract":true,"cache":false},{"value":"http://manifestserv.paypalcorp.com/containerspecs/specid/genesisconfig-v2.3.5-apirouter/config.zip","extract":true,"cache":false}]},"assignedPorts":{},"instanceId":0},"status":"PENDING","failureCount":1,"taskEvents":[{"timestamp":1655896691621,"status":"PENDING","message":"Rescheduled","scheduler":"geninmm01006.gpf-dev.us-central1.gcp.dev.paypalinc.com"}],"ancestorId":"admin-genesisinfra-apirouter-a-0-66eddb01-48d9-4e02-b37c-9652d55e139a","timesPartitioned":0},{"assignedTask":{"taskId":"admin-clustercustodian-genesisinfra-dockerserv-a-0-8c850f1a-4aa1-46a4-b525-3589b131daf0","slaveId":"","slaveHost":"","task":{"isService":true,"priority":0,"maxTaskFailures":0,"owner":{"user":"admin"},"production":false,"constraints":[{"name":"dedicated","constraint":{"value":{"negated":false,"values":["admin/adminjob"]}}}],"executorConfig":{"name":"dce-go-paypal-ipvlan","data":""},"metadata":[{"key":"genesisapi.env.pp_az","value":"gcp_MSMASTER"},{"key":"genesisapi.env.ENV_YAM_NAMESPACE","value":"m2mtIgC"},{"key":"com.genesis.hostName","value":""},{"key":"com.genesis.namespace","value":"clustercustodian-genesisinfra"},{"key":"genesisapi.env.METADATA_ENV_FILE","value":"additionalConfigs={}\napplicationProfilingTools=\nclassOfService=QA\ncronusType=applicationService\ninstance.az=lvs03\ninstance.azLabel=\ninstance.datacenter=lvs03\ninstance.fqdn=te-alm-48095402108340824489929.qa.paypal.com\ninstance.region=LVS\ninstance.topoId=unknown\ninstance.vpc=DEV\nlabel=dockerserv\nnewManifestId=dockerserv-0.0.1_SNAPSHOT_20211020143115891\nodbMoName=te-alm-48095402108340824489929\norganizationCode=ZPC40000\norganizationName=PayPal\npaasRealm=QA\nserviceName=dockerserv-app:CLDte-alm-48095402108340824489929\nservicePath=/\nstackId=raptor-docker.service\nsvc.archTier=app\nsvc.architecture=r1\nsvc.nugget=dockerserv\nsvc.userData="},{"key":"genesisapi.env.amq_namespace","value":"m2mtIgC"},{"key":"resourceLimit.gpu.required","value":"false"},{"key":"keymaker_url","value":"https://keymakerapi-vip.qa.paypal.com"},{"key":"fileName","value":"docker-compose-base.yml,docker-compose-extends-base.yml,docker-compose-qa.yml,docker-compose-extends-qa.yml,docker-compose-healthcheck.yml"},{"key":"com.genesis.containerIP","value":""},{"key":"com.admin.dockerserv.svcport","value":"18999"},{"key":"com.admin.dockerserv.namespace","value":"clustercustodian-genesisinfra"},{"key":"genesisapi.env.ENV_NAMESPACE","value":"m2mtIgC"},{"key":"com.genesis.nsvip","value":"10.180.82.131"},{"key":"dz","value":"genesisinfra"},{"key":"genesisapi.env.tenant","value":"admin"},{"key":"genesisapi.env.APPNAME","value":"dockerserv"},{"key":"clustername","value":"avi-controller"},{"key":"com.genesis.version","value":"dockerserv-0.0.1_SNAPSHOT_20211020143115891"},{"key":"com.admin.dockerserv.pooluuid","value":"pool-dfb83682-2a7c-4e1b-b0a7-284e0bba8494"},{"key":"genesisapi.env.METADATA_JSON_FILE","value":"{\n\"additionalConfigs\": {},\n\"applicationProfilingTools\": \"\",\n\"classOfService\": \"QA\",\n\"cronusType\": \"applicationService\",\n\"instance.az\": \"lvs03\",\n\"instance.azLabel\": \"\",\n\"instance.datacenter\": \"lvs03\",\n\"instance.fqdn\": \"te-alm-48095402108340824489929.qa.paypal.com\",\n\"instance.region\": \"LVS\",\n\"instance.topoId\": \"unknown\",\n\"instance.vpc\": \"DEV\",\n\"label\": \"dockerserv\",\n\"newManifestId\": \"dockerserv-0.0.1_SNAPSHOT_20211020143115891\",\n\"odbMoName\": \"te-alm-48095402108340824489929\",\n\"organizationCode\": \"ZPC40000\",\n\"organizationName\": \"PayPal\",\n\"paasRealm\": \"QA\",\n\"serviceName\": \"dockerserv-app:CLDte-alm-48095402108340824489929\",\n\"servicePath\": \"/\",\n\"stackId\": \"raptor-docker.service\",\n\"svc.archTier\": \"app\",\n\"svc.architecture\": \"r1\",\n\"svc.nugget\": \"dockerserv\",\n\"svc.userData\": \"\"\n}"},{"key":"secretProvider","value":"keymaker"},{"key":"com.genesis.dz","value":"genesisinfra"},{"key":"genesisapi.env.USERSTAGE_NAME","value":"m2mtIgC"},{"key":"com.genesis.slbpool","value":"true"},{"key":"genesisapi.env.pp_geo","value":"MSMASTER"},{"key":"com.admin.dockerserv.version","value":"dockerserv-0.0.1_SNAPSHOT_20211020143115891"},{"key":"fetch_token","value":"true"},{"key":"deployStrategy","value":"DEPLOY_STRATEGY_ROLLING_UPDATE"},{"key":"version","value":"dockerserv-0.0.1_SNAPSHOT_20211020143115891"},{"key":"com.admin.poolname","value":"dockerserv"},{"key":"com.genesis.svcport","value":"18999"},{"key":"genesisapi.env.HAPROXY_JSON","value":"{\"isProxyNode\":false,\"proxyHost\":\"te-alm-48095402108340824489929.qa.paypal.com\"}"},{"key":"nsvip","value":"10.180.82.131"},{"key":"requuid","value":"b162c7fe-513e-11ec-9321-42010ab44953"},{"key":"keymaker_skipverify","value":"1"},{"key":"namespace","value":"clustercustodian-genesisinfra"},{"key":"com.genesis.cluster","value":"genesisinfra"},{"key":"genesisapi.env.SVCPORT","value":"18999"},{"key":"com.genesis.pool","value":"dockerserv"},{"key":"genesisapi.env.namespace","value":"clustercustodian-genesisinfra"},{"key":"genesisapi.env.pp_datacenter","value":"gcp_MSMASTER"},{"key":"tenant","value":"admin"},{"key":"pool","value":"dockerserv"},{"key":"com.genesis.tenant","value":"admin"},{"key":"geo","value":""},{"key":"com.admin.dockerserv.poolservercount","value":"1"},{"key":"genesisapi.env.pp_region","value":"none"},{"key":"genesisapi.env.pool","value":"dockerserv"}],"job":{"role":"admin","environment":"clustercustodian-genesisinfra","name":"dockerserv-a"},"container":{"mesos":{}},"tier":"preemptible","resources":[{"numCpus":0.9},{"ramMb":100},{"diskMb":100},{"namedPort":"org.apache.aurora.port.0"},{"namedPort":"org.apache.aurora.port.1"},{"namedPort":"org.apache.aurora.port.2"},{"namedPort":"org.apache.aurora.port.3"},{"namedPort":"org.apache.aurora.port.4"}],"mesosFetcherUris":[{"value":"http://manifestserv.paypalcorp.com/containerspecs/dockerserv-0.0.1_SNAPSHOT_20211020143115891/yaml","extract":true,"cache":false}]},"assignedPorts":{},"instanceId":0},"status":"PENDING","failureCount":57235,"taskEvents":[{"timestamp":1656320989434,"status":"THROTTLED","message":"Rescheduled, penalized for 300000 ms for flapping","scheduler":"geninmm01006.gpf-dev.us-central1.gcp.dev.paypalinc.com"},{"timestamp":1656321289434,"status":"PENDING","scheduler":"geninmm01006.gpf-dev.us-central1.gcp.dev.paypalinc.com"}],"ancestorId":"admin-clustercustodian-genesisinfra-dockerserv-a-0-173b39cb-2199-4740-bb88-051ce44cf8f2","timesPartitioned":0},{"assignedTask":{"taskId":"admin-genesisinfra-apirouter-a-2-2ae1ae4a-c21c-4e71-8f91-fb14381022dd","slaveId":"","slaveHost":"","task":{"isService":true,"priority":0,"maxTaskFailures":0,"owner":{"user":"admin"},"production":false,"constraints":[{"name":"host","constraint":{"limit":{"limit":1}}},{"name":"dedicated","constraint":{"value":{"negated":false,"values":["admin/adminjob"]}}}],"executorConfig":{"name":"dce-go-paypal-host","data":""},"metadata":[{"key":"com.genesis.version","value":"genesisapiserv-v3.11.4-v2.3.5"},{"key":"genesisapi.env.SVCPORT","value":"8100"},{"key":"com.genesis.slbpool","value":"true"},{"key":"com.admin.poolname","value":"apirouter"},{"key":"com.admin.apirouter.svcport","value":"8100"},{"key":"com.genesis.pool","value":"apirouter"},{"key":"com.genesis.hostName","value":""},{"key":"com.genesis.svcport","value":"8100"},{"key":"version","value":"genesisapiserv-v3.11.4-v2.3.5"},{"key":"com.admin.apirouter.poolservercount","value":"3"},{"key":"fetch_token","value":"true"},{"key":"deployStrategy","value":"DEPLOY_STRATEGY_ROLLING_UPDATE"},{"key":"com.admin.apirouter.version","value":"genesisapiserv-v3.11.4-v2.3.5"},{"key":"genesisapi.env.FSM_COMPOSE_FILENAMES","value":"docker-compose.yaml,config/docker-compose-base.yml,config/docker-compose-extends-base.yml"},{"key":"resourceLimit.gpu.required","value":"false"},{"key":"fileName","value":"docker-compose.yml,config/docker-compose-base.yml,config/docker-compose-extends-base.yml"},{"key":"keymaker_url","value":"https://keymakerapi-vip.qa.paypal.com"},{"key":"genesisapi.env.pool","value":"apirouter"},{"key":"genesisapi.env.DZENV","value":"genesisinfra"},{"key":"com.genesis.containerIP","value":""},{"key":"requuid","value":"93130a55-cbb4-11eb-9f10-42010ab4496b"},{"key":"namespace","value":"genesisinfra"},{"key":"keymaker_skipverify","value":"1"},{"key":"nsvip","value":"10.180.82.129"},{"key":"dz","value":"genesisinfra"},{"key":"genesisapi.env.DZ","value":"gcp"},{"key":"genesisapi.env.tenant","value":"admin"},{"key":"pool","value":"apirouter"},{"key":"com.admin.apirouter.namespace","value":"genesisinfra"},{"key":"clustername","value":"avi-controller"},{"key":"com.genesis.namespace","value":"genesisinfra"},{"key":"genesisapi.env.FSM_RESOURCE_URLS","value":"http://genesisinfra.qa.paypal.com:8100/config/fsm/docker-compose.yaml http://manifestserv.paypalcorp.com/containerspecs/specid/genesisconfig-v2.3.5-fsm/config.zip"},{"key":"com.genesis.cluster","value":"genesisinfra"},{"key":"com.admin.apirouter.pooluuid","value":"pool-3d9ffffe-a7aa-40f2-8e9c-7126c75690b2"},{"key":"com.genesis.nsvip","value":"10.180.82.129"},{"key":"secretProvider","value":"keymaker"},{"key":"tenant","value":"admin"},{"key":"genesisapi.env.namespace","value":"genesisinfra"},{"key":"com.genesis.dz","value":"genesisinfra"},{"key":"com.genesis.tenant","value":"admin"},{"key":"geo","value":""}],"job":{"role":"admin","environment":"genesisinfra","name":"apirouter-a"},"container":{"mesos":{}},"tier":"preemptible","resources":[{"ramMb":1024},{"numCpus":1},{"diskMb":100},{"namedPort":"org.apache.aurora.port.0"},{"namedPort":"org.apache.aurora.port.1"},{"namedPort":"org.apache.aurora.port.2"},{"namedPort":"org.apache.aurora.port.3"},{"namedPort":"org.apache.aurora.port.4"}],"mesosFetcherUris":[{"value":"http://manifestserv.paypalcorp.com/containerspecs/genesisapiserv-v3.11.4/yaml","extract":true,"cache":false},{"value":"http://manifestserv.paypalcorp.com/containerspecs/specid/genesisconfig-v2.3.5-apirouter/config.zip","extract":true,"cache":false}]},"assignedPorts":{},"instanceId":2},"status":"PENDING","failureCount":4252,"taskEvents":[{"timestamp":1656321848345,"status":"PENDING","message":"Rescheduled","scheduler":"geninmm01006.gpf-dev.us-central1.gcp.dev.paypalinc.com"}],"ancestorId":"admin-genesisinfra-apirouter-a-2-5da623a6-394e-44ff-af61-62b7f55882e5","timesPartitioned":0},{"assignedTask":{"taskId":"admin-clustercustodian-genesisinfra-custodian-a-0-23aa2166-6c27-4c7e-8a03-cdd2a7296075","slaveId":"","slaveHost":"","task":{"isService":true,"priority":0,"maxTaskFailures":0,"owner":{"user":"admin"},"production":false,"constraints":[{"name":"dedicated","constraint":{"value":{"negated":false,"values":["admin/adminjob"]}}}],"executorConfig":{"name":"dce-go-paypal-host","data":""},"metadata":[{"key":"com.genesis.pool","value":"custodian"},{"key":"com.genesis.slbpool","value":"true"},{"key":"version","value":"custodian-v1.7.4a"},{"key":"com.genesis.hostName","value":""},{"key":"com.admin.poolname","value":"custodian"},{"key":"com.genesis.namespace","value":"clustercustodian-genesisinfra"},{"key":"com.admin.custodian.namespace","value":"clustercustodian-genesisinfra"},{"key":"fetch_token","value":"true"},{"key":"genesisapi.env.METADATA_ENV_FILE","value":"additionalConfigs={}\napplicationProfilingTools=\nclassOfService=QA\ncronusType=applicationService\ninstance.az=gcp\ninstance.azLabel=gcp\ninstance.datacenter=gcp\ninstance.fqdn=bidetest.qa.paypal.com\ninstance.region=gcp\ninstance.topoId=unknown\ninstance.vpc=QA\nlabel=custodian\nnewManifestId=custodian-rc1-af84df7\nodbMoName=bidetest\norganizationCode=ZPC40000\norganizationName=PayPal\npaasRealm=QA\nserviceName=custodian-app:CLDbidetest\nservicePath=/\nstackId=custom-docker\nsvc.archTier=mid-tier\nsvc.architecture=custom-mid\nsvc.nugget=custodian\nsvc.userData="},{"key":"requuid","value":"1ebde306-99af-11ec-8f87-42010ab44969"},{"key":"com.genesis.version","value":"custodian-v1.7.4a"},{"key":"deployStrategy","value":"DEPLOY_STRATEGY_ROLLING_UPDATE"},{"key":"resourceLimit.gpu.required","value":"false"},{"key":"com.admin.custodian.svcport","value":"9188"},{"key":"com.admin.custodian.pooluuid","value":"pool-e2ac17c1-a6a7-466a-88d5-a8ead73922eb"},{"key":"keymaker_url","value":"https://keymakerapi-vip.qa.paypal.com"},{"key":"genesisapi.env.DZENV","value":"genesisinfra"},{"key":"com.genesis.containerIP","value":""},{"key":"nsvip","value":"10.180.82.131"},{"key":"genesisapi.env.SVCPORT","value":"9188"},{"key":"genesisapi.env.METADATA_JSON_FILE","value":"{\n\"additionalConfigs\": null,\n\"applicationProfilingTools\": \"\",\n\"classOfService\": \"QA\",\n\"cronusType\": \"applicationService\",\n\"instance.az\": \"gcp\",\n\"instance.azLabel\": \"gcp\",\n\"instance.datacenter\": \"gcp\",\n\"instance.fqdn\": \"bidetest.qa.paypal.com\",\n\"instance.region\": \"gcp\",\n\"instance.topoId\": \"unknown\",\n\"instance.vpc\": \"QA\",\n\"label\": \"custodian\",\n\"newManifestId\": \"custodian-rc1-af84df7\",\n\"odbMoName\": \"bidetest\",\n\"organizationCode\": \"ZPC40000\",\n\"organizationName\": \"PayPal\",\n\"paasRealm\": \"QA\",\n\"serviceName\": \"custodian-app:CLDbidetest\",\n\"servicePath\": \"/\",\n\"stackId\": \"custom-docker\",\n\"svc.archTier\": \"mid-tier\",\n\"svc.architecture\": \"custom-mid\",\n\"svc.nugget\": \"custodian\",\n\"svc.userData\": \"\"\n}"},{"key":"keymaker_skipverify","value":"1"},{"key":"namespace","value":"clustercustodian-genesisinfra"},{"key":"com.genesis.nsvip","value":"10.180.82.131"},{"key":"dz","value":"genesisinfra"},{"key":"genesisapi.env.DZ","value":"gcp"},{"key":"genesisapi.env.tenant","value":"admin"},{"key":"com.genesis.svcport","value":"9188"},{"key":"clustername","value":"avi-controller"},{"key":"genesisapi.env.GENESISCONFIG_PATH","value":"gcp/genesisinfra"},{"key":"genesisapi.env.APPNAME","value":"custodian"},{"key":"com.admin.custodian.version","value":"custodian-v1.7.4a"},{"key":"com.genesis.cluster","value":"genesisinfra"},{"key":"genesisapi.env.namespace","value":"clustercustodian-genesisinfra"},{"key":"secretProvider","value":"keymaker"},{"key":"tenant","value":"admin"},{"key":"com.genesis.dz","value":"genesisinfra"},{"key":"com.genesis.tenant","value":"admin"},{"key":"geo","value":""},{"key":"fileName","value":""},{"key":"genesisapi.env.pool","value":"custodian"},{"key":"pool","value":"custodian"},{"key":"com.admin.custodian.poolservercount","value":"1"}],"job":{"role":"admin","environment":"clustercustodian-genesisinfra","name":"custodian-a"},"container":{"mesos":{}},"tier":"preemptible","resources":[{"ramMb":8192},{"numCpus":0.9},{"diskMb":100},{"namedPort":"org.apache.aurora.port.0"},{"namedPort":"org.apache.aurora.port.1"},{"namedPort":"org.apache.aurora.port.2"},{"namedPort":"org.apache.aurora.port.3"},{"namedPort":"org.apache.aurora.port.4"}],"mesosFetcherUris":[{"value":"http://manifestserv.paypalcorp.com/containerspecs/custodian-v1.7.4a/yaml","extract":true,"cache":false}]},"assignedPorts":{},"instanceId":0},"status":"PENDING","failureCount":1475,"taskEvents":[{"timestamp":1656321848337,"status":"PENDING","message":"Rescheduled","scheduler":"geninmm01006.gpf-dev.us-central1.gcp.dev.paypalinc.com"}],"ancestorId":"admin-clustercustodian-genesisinfra-custodian-a-0-2626a330-69b6-4a1e-a619-ed7608da2f3f","timesPartitioned":0}]}

Reply:
this is what I am worried because we will print too many things out and most of them are unnecessary. Lets just print out the list of full task id.

Reply:
/private/var/folders/p4/xg62qpjd17jc8wd_vsv8zww80000gp/T/GoLand/___go_build_main_go fetch tasks -z geninmm01005.gpf-dev.us-central1.gcp.dev.paypalinc.com -x pending --toJSON
INFO[0000] Scheduler URL from ZK:  http://geninmm01006.gpf-dev.us-central1.gcp.dev.paypalinc.com:8081 
INFO[0000] gorealis clientConfig url: http://geninmm01006.gpf-dev.us-central1.gcp.dev.paypalinc.com:8081/api 
INFO[0000] Fetching tasks for environment/role/job:[//]  
INFO[0000] Fetching tasks for a given status: pending   
{"PENDING":["admin-clustercustodian-genesisinfra-dockerserv-a-0-8c850f1a-4aa1-46a4-b525-3589b131daf0","admin-genesisinfra-apirouter-a-0-8355edb8-c991-476e-9373-e325a25e6e74","admin-clustercustodian-genesisinfra-custodian-a-0-23aa2166-6c27-4c7e-8a03-cdd2a7296075","admin-genesisinfra-apirouter-a-1-3b8122aa-35b2-4ebb-8d58-e0fa937711b6","admin-genesisinfra-apirouter-a-2-2ae1ae4a-c21c-4e71-8f91-fb14381022dd"]}